src/utils/BatchManager.js

```diff
@@ -138,32 +129,42 @@ class BatchManager {
    * job context. Additionally, duration is calculated.
    */
   render(token) {
-    const start = now();
     const context = this.jobContexts[token];
     const name = context.name;
 
     const { getComponent } = this.config;
 
+    const timeStartGetComponent = process.hrtime();
     const result = getComponent(name, context);
 
     return Promise.resolve(result).then((renderFn) => {
+      const timeEndGetComponent = process.hrtime(timeStartGetComponent);
       // ensure that we have this component registered
       if (!renderFn || typeof renderFn !== 'function') {
         // component not registered
         context.statusCode = 404;
-        context.duration = msSince(start);
+        context.duration = {
+          getComponent: timeEndGetComponent,
+          render: null,
+        };
         throw notFound(name);
       }
 
       let response = null;
 
+      const timeStartRender = process.hrtime();
+
       // render the component!
       try {
         context.html = renderFn(context.props);
       } catch (e) {
         response = Promise.reject(e);
       } finally {
-        context.duration = msSince(start);
+        const timeEndRender = process.hrtime(timeStartRender);
+        context.duration = {
```

Comment:
I think an object makes sense, but I'm not so sure about the name anymore. I think `timing` might be more accurate.

It might be worth looking at [`PerformanceTiming`](https://developer.mozilla.org/en-US/docs/Web/API/PerformanceTiming) for inspiration.